src/interpreter.c

```diff
@@ -25,7 +25,48 @@ typedef struct {
     int continue_at; // statement index to jump to after leaving exception handler (0 if none)
 } interpreter_state;
 
-#include "interpreter-stacktrace.c"
+
+// general alloca rules are incompatible on C and C++, so define a macro that deals with the difference
+#ifdef __cplusplus
+#define JL_CPPALLOCA(var,n)                                                         \
+  var = (decltype(var))alloca((n))
+#else
+#define JL_CPPALLOCA(var,n)                                                         \
+  JL_GCC_IGNORE_START("-Wc++-compat")                                               \
+  var = alloca((n));                                                                \
+  JL_GCC_IGNORE_STOP
+#endif
+
+#ifdef __clang_analyzer__
+
+extern void JL_GC_ENABLEFRAME(interpreter_state*) JL_NOTSAFEPOINT;
+
+// This is necessary, because otherwise the analyzer considers this undefined
+// behavior and terminates the exploration
+#define JL_GC_PUSHFRAME(frame,n)     \
+  JL_CPPALLOCA(frame, sizeof(*frame)+((n) * sizeof(jl_value_t*)));                  \
+  memset(&frame[1], 0, sizeof(void*) * n); \
+  _JL_GC_PUSHARGS((jl_value_t**)&frame[1], n);
+
+#else
+
+#define JL_GC_ENCODE_PUSHFRAME(n)  ((((size_t)(n))<<2)|2)
+
+#define JL_GC_PUSHFRAME(frame,n)                                                    \
+  JL_CPPALLOCA(frame, sizeof(*frame)+(((n)+3)*sizeof(jl_value_t*)));                \
+  ((void**)&frame[1])[0] = NULL;                                                    \
+  ((void**)&frame[1])[1] = (void*)JL_GC_ENCODE_PUSHFRAME(n);                        \
+  ((void**)&frame[1])[2] = jl_pgcstack;                                             \
+  memset(&((void**)&frame[1])[3], 0, (n)*sizeof(jl_value_t*));                      \
+  jl_pgcstack = (jl_gcframe_t*)&(((void**)&frame[1])[1])
```

Comment:
Nice resolution.

I do think this makes a lot more sense than putting the space for the frame inside `interpreter_state` :+1: